### PR TITLE
counting headers in __frang_http_field_len() manually

### PR DESCRIPTION
**Pull request by Mihail-Makei**
Preliminary changes (counting the headers manually using table)

### CODE REVIEW

File: fw/t/unit/http_msg.c [hunk 1]
```diff
@@ -1 +0,0 @@
-../../http_msg.c
```
Comment (krizhanovsky) [inline fw/t/unit/http_msg.c:1]: Please revert the movement of source files to the unit tests directory

File: fw/http_limits.c [hunk 2]
```diff
@@ -342,15 +342,17 @@ frang_http_uri_len(const TfwHttpReq *req, FrangAcc *ra, unsigned int uri_len)
...
 		       unsigned int field_len,  unsigned int hdr_cnt)
 {
 	const TfwStr *field, *end, *dup, *dup_end;
 
-	if (hdr_cnt && req->h_tbl->off >= hdr_cnt) {
-		frang_limmsg("HTTP headers number", req->h_tbl->off,
+	if (hdr_cnt && req->h_tbl->cnt > hdr_cnt) {
+		frang_limmsg("HTTP headers number", req->h_tbl->cnt,
 			     hdr_cnt, &FRANG_ACC2CLI(ra)->addr);
+
```
Comment (krizhanovsky) [inline fw/http_limits.c:355]: Here and in line 345 are senseless changes of just adding 2 empty lines

File: fw/http_parser.h [hunk 3]
```diff
@@ -26,6 +26,7 @@
 typedef struct {
 	unsigned int	size;	/* number of elements in the table */
 	unsigned int	off;
+	unsigned int	cnt;
```
Comment (krizhanovsky) [inline fw/http_parser.h:29]: Probably we'll have 4 bytes hole here, so I'm wondering if we do support 2G headers and can't we just use `unsigned short` at least for `cnt` and `off`?

**Review by krizhanovsky — CHANGES_REQUESTED** [review]
At least removing the changes for the source files copying is required. Also there are couple of cleanups/improvements.

File: fw/http_limits.c (hunk 2 above)
Comment (Mihail-Makei) [reply fw/http_limits.c:355]: Fixed

File: fw/http_parser.h (hunk 3 above)
Comment (Mihail-Makei) [reply fw/http_parser.h:29]: No problem provided we do not need bigger number of headers

Fixed

File: fw/t/unit/http_msg.c (hunk 1 above)
Comment (Mihail-Makei) [reply fw/t/unit/http_msg.c:1]: Hmmm, file seems to exist in unit tests directory

**Review by const-t — APPROVED** [review]
LGTM. Please do squash merge.

**Pull request merged**